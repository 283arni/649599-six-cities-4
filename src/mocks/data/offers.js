export const offers = [
  {
    id: 1,
    photo: `./img/apartment-01.jpg`,
    premium: true,
    price: 130,
    title: `Nice, cozy, warm big bed apartment`,
    type: `apartment`,
    rating: 3
  }, {
    id: 2,
    photo: `./img/apartment-02.jpg`,
    premium: true,
    price: 250,
    title: `Wood and stone place`,
    type: `room`,
    rating: 4.9
  }, {
    id: 3,
    photo: `./img/apartment-03.jpg`,
    premium: false,
    price: 220,
    title: `Beautiful &amp; luxurious apartment at great location`,
    type: `hotel`,
    rating: 3.7
  }, {
    id: 4,
    photo: `./img/apartment-01.jpg`,
    premium: false,
    price: 120,
    title: `Wood and stone place`,
    type: `apartment`,
    rating: 3
  }
];

export const property = {
  photos: [`./img/apartment-01.jpg`, `./img/apartment-02.jpg`, `./img/apartment-03.jpg`, `./img/studio-01.jpg`, `./img/apartment-03.jpg`],
  title: `Beautiful &amp; luxurious apartment at great location`,
  description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`],
  premium: false,
  type: {
    room: `Private Room`
  },
  rating: 3.7,
  amountBedrooms: 3,
  maxGustes: 3,
  price: 120,
  things: [`WiFi`, `Cable TV`, `Conditioner`],
  owner: {
    avatar: `./img/avatar-angelina.jpg`,
    name: `Petiy`,
    super: true
  }
};
