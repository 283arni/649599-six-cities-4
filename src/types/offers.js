import PropTypes from 'prop-types';


export const offerType = {
  amountBedrooms: PropTypes.number.isRequired,
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    coords: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
    zoom: PropTypes.number.isRequired
  }).isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  things: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  owner: PropTypes.shape({
    avatar: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    super: PropTypes.bool.isRequired,
    id: PropTypes.number.isRequired
  }).isRequired,
  photos: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  favorite: PropTypes.bool.isRequired,
  premium: PropTypes.bool.isRequired,
  coords: PropTypes.shape({
    target: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
    zoom: PropTypes.number.isRequired
  }).isRequired,
  maxGustes: PropTypes.number.isRequired,
  photo: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export const favoriteOfferType = {
  city: PropTypes.string.isRequired,
  offers: PropTypes.arrayOf(
      PropTypes.shape(offerType)
  ).isRequired
};
