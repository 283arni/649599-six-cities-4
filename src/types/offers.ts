

export interface offerType {
  amountBedrooms: number;
  city: {
    name: string;
    coords: number[];
    zoom: number;
  };
  description: string;
  id: number;
  things: string[];
  owner: {
    avatar: string;
    name: string;
    super: boolean;
    id: number;
  };
  photos: string[];
  favorite: boolean;
  premium: boolean;
  coords: {
    target: number[];
    zoom: number;
  };
  maxGustes: number;
  photo: string;
  price: number;
  rating: number;
  title: string;
  type: string;
};

export interface favoriteOfferType {
  city: string;
  offers: offerType[];
};
