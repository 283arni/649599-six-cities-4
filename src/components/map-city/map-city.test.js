import React from 'react';
import renderer from 'react-test-renderer';
import MapCity from './map-city';
import {offers, offer} from '../../mocks/test/offers';


jest.mock(`leaflet`, () => ({
  icon: jest.fn(),
  map: jest.fn().mockReturnValue({
    setView: jest.fn(),
    remove: jest.fn()
  }),
  tileLayer: jest.fn().mockReturnValue({
    addTo: jest.fn()
  }),
  marker: jest.fn().mockReturnValue({
    addTo: jest.fn()
  }),
}));

it(`check render MapCity`, () => {
  const component = renderer.create(
      <MapCity
        offers={offers}
        currentCity={offer.city.name}
        cityOffers={offer.city}
      />, {
        createNodeMock: () => document.createElement(`div`)
      }
  ).toJSON();

  expect(component).toMatchSnapshot();
});
