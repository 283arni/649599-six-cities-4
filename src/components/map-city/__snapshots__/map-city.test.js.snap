// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render MapCity 1`] = `
<div
  className="cities__right-section"
>
  <section
    className="cities__map map"
  >
    <div
      id="map"
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </section>
</div>
`;
